tests/core/channels/test_twilio_voice.py

```diff
@@ -0,0 +1,215 @@
+import json
+import logging
+
+import pytest
+from sanic.request import Request
+
+import rasa.core
+from rasa.core import utils
+from rasa.shared.exceptions import InvalidConfigException
+from rasa.core.channels.twilio_voice import TwilioVoiceInput
+from rasa.core.channels.twilio_voice import TwilioVoiceCollectingOutputChannel
+
+logger = logging.getLogger(__name__)
+
+
+async def test_twilio_voice_twiml_response_text():
+
+    inputs = {
+        "initial_prompt": "hello",
+        "reprompt_fallback_phrase": "i didn't get that",
+        "speech_model": "default",
+        "speech_timeout": "5",
+        "assistant_voice": "woman",
+        "enhanced": "false"
+    }
+
+    tv = TwilioVoiceInput(**inputs)
+
+    output_channel = TwilioVoiceCollectingOutputChannel()
+
+    await output_channel.send_text_message(recipient_id="Chuck Norris", text="Test:")
+    assert len(output_channel.messages) == 1
+    assert output_channel.messages[0]["text"] == "Test:"
+
+    twiml = tv.build_twilio_voice_response(output_channel.messages)
+    assert (
+        str(twiml)
+        == '<?xml version="1.0" encoding="UTF-8"?><Response><Gather action="/webhooks/twilio_voice/webhook" actionOnEmptyResult="true" enhanced="false" input="speech" speechModel="default" speechTimeout="5"><Say voice="woman">Test:</Say></Gather></Response>'
+    )
+
+
+async def test_twilio_voice_twiml_response_buttons():
+
+    inputs = {
+        "initial_prompt": "hello",
+        "reprompt_fallback_phrase": "i didn't get that",
+        "speech_model": "default",
+        "speech_timeout": "5",
+        "assistant_voice": "woman",
+        "enhanced": "false"
+    }
+
+    tv = TwilioVoiceInput(**inputs)
+
+    output_channel = TwilioVoiceCollectingOutputChannel()
+    await output_channel.send_text_with_buttons(
+        recipient_id="Chuck Norris",
+        text="Buttons:",
+        buttons=[
+            {"title": "Yes", "payload": "/affirm"},
+            {"title": "No", "payload": "/deny"},
+        ],
+    )
+    assert len(output_channel.messages) == 3
+    message_str = " ".join([m["text"] for m in output_channel.messages])
+    assert message_str == "Buttons: Yes No"
+
+    twiml = tv.build_twilio_voice_response(output_channel.messages)
+    assert (
+        str(twiml)
+        == '<?xml version="1.0" encoding="UTF-8"?><Response><Say voice="woman">Buttons:</Say><Pause length="1" /><Say voice="woman">Yes</Say><Pause length="1" /><Gather action="/webhooks/twilio_voice/webhook" actionOnEmptyResult="true" enhanced="false" input="speech" speechModel="default" speechTimeout="5"><Say voice="woman">No</Say></Gather></Response>'
+    )
+
+
+async def test_twilio_invalid_assistant_voice():
```

Comment:
why are these tests `async` if we don't use `await` in them? 

I'd also consider using https://docs.pytest.org/en/stable/parametrize.html to group all invalid config tests! 

Nice and thorough testing btw 🎉

Reply:
Oh, that's much more extensible wrapping those config tests together instead of creating separate functions for each!